frontend/src/App.tsx

```diff
@@ -378,6 +423,7 @@ function App(props: { setOnDisconnect: Dispatch<SetStateAction<Callback | undefi
   const [currentMazeCompletionList, setCurrentMazeCompletionList] = useState<MazeCompletionInfo[]>(
     [],
   );
+  const {emitFinishGame} = appState;
```

Comment:
If you're going to de-structure here, we should do it for everything in appState or not at all
ie:
appState.apiClient
appState.sessionToken
etc.

Reply:
I see your issue, it wants to putt appState in the dependency array :(